mirdata/datasets/egfxset.py

```diff
@@ -0,0 +1,383 @@
+
+    @property
+    def knob_names(self):
+        return self._track_metadata.get("Knob Names")
+
+    @property
+    def knob_type(self):
+        return self._track_metadata.get("Knob Type")
+
+    @property
+    def setting(self):
+        return self._track_metadata.get("Setting")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """Solo guitar audio (mono)
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, metadata=self._track_metadata
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load EGFxSet guitar audio
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+
+    Returns:
+        * np.ndarray - audio signal
+        * float - sample rate
+
+    """
+    return librosa.load(fhandle, sr=None, mono=True)
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The EGFxSet dataset
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="egfxset",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(self.data_home, "egfxset_metadata.csv")
+        metadata_index = {}
+        tracknames = self.track_ids
+        cuerdas = {"1": 64, "2": 59, "3": 55, "4": 50, "5": 45, "6": 40}
+
+        file_handle = open(metadata_path, "r")
+        reader = list(csv.DictReader(file_handle))
+
+        indexname = []
+        for name in reader:
+            indexname.append(name["Effect "].split(" ")[0])
+
+        for track in tracknames:
+
+            if track[:3] == "RAT":
+                trackiden = track[:3].lower()
+                trackiden = trackiden.replace(" ", "")
+                trackiden = trackiden.replace("_", "")
+
+            if track[:3] != "RAT":
+                trackiden = re.findall("[A-Z][^A-Z]*", track)[0].lower()
+                trackiden = trackiden.replace(" ", "")
+                trackiden = trackiden.replace("_", "")
+
+            noteCord = track.split("/")[1].split("-")
+            if len(noteCord) != 2:
+                noteCord = track.split("/")[1].split(".")
+
+            if track[:2].lower() == "cl":
+                metadata_index[track] = {
+                    "String-fret Tuple": [
+                        int(s) for s in re.findall(r"\b\d+\b", track.split("/")[1])
```

Comment:
Is this "/" part of a path you build, or is it harcoded in the metadata? I'm asking this to make sure that this code would run on Windows as well (see my comment in the tests)

Reply:
Our track_ids use this sign "/" (i.e. Clean_Bridge/6-0), we take the string and split it to only retrieve the numbers at the end of the id, we are not working here with any path, only a string.